File: csharp/card-game-war/src/game.cs
namespace card_game_war;

using System.Collections.Generic;

public class game {

    public enum Suit {
        Spade,
        Club,
        Diamond,
        Heart
    }

    public enum Rank {
        Two = 2,
        Three = 3,
        Four = 4,
        Five = 5,
        Six = 6,
        Seven = 7,
        Eight = 8,
        Nine = 9,
        Ten = 10,
        Jack,
        Queen,
        King,
        Ace
    }

    public readonly record struct Card(Suit suit, Rank rank);

    /*
    public static List<Card> deck() {
        List<Card> deck = new();
        foreach (Suit suit in Enum.GetValues(typeof(Suit))) {
            foreach (Rank rank in Enum.GetValues(typeof(Rank))) {
                deck.Add(new Card(suit, rank));
            }
        }
        return deck;
    }
    */

    public static void playRound (Card player1_card, Card player2_card) =>
        throw new System.NotImplementedException();

    public static void playGame (List<Card> player1_cards, List<Card> player2_cards) =>
        throw new System.NotImplementedException();
}

File: fsharp/card-game-war/src/game.fs
namespace card_game_war

module game =

    // feel free to use these cards or use your own data structure

    type Suit =
        | Spade
        | Club
        | Diamond
        | Heart

    type Rank =
        | Value of int
        | Jack
        | Queen
        | King
        | Ace

    type Card =
        { Suit : Suit; Rank : Rank }
        static member Create(suit : Suit, rank : Rank) =
            { Suit = suit; Rank = rank }

    (*
    let suits = [ Spade; Club; Diamond; Heart ]
    let heads = [ Jack; Queen; King; Ace ]

    let ranks =
        [   for v in 2 .. 10 -> Value v
            for head in heads -> head
        ]

    let deck =
        seq {
            for suit in suits do
                for rank in ranks -> Card.Create(suit, rank)
        }
    *)

    let playRound (player1_card : Card) (player2_card : Card) =
        raise <| System.NotImplementedException()

    let playGame (player1_cards : Card list) (player2_cards : Card list) =
        raise <| System.NotImplementedException()
